File: MtxVec.Delphi/FloatEdit1.pas
unit FloatEdit1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Basic2, MtxComCtrls, StdCtrls, ComCtrls, ExtCtrls, Math387;

type
  TfrmFloatEdit1 = class(TBasicForm2)
    MtxFloatEdit1: TMtxFloatEdit;
    MtxFloatEdit2: TMtxFloatEdit;
    Label1: TLabel;                                
    Label2: TLabel;
    Label3: TLabel;
    Button1: TButton;
    Button2: TButton;
    Button3: TButton;
    Button4: TButton;
    Edit1: TEdit;
    Label4: TLabel;
    Button5: TButton;
    Button6: TButton;
    procedure Button1Click(Sender: TObject);
    procedure Button5Click(Sender: TObject);
    procedure Button6Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
    Res: TCplx;
    Z1,Z2: TCplx;
  public
    { Public declarations }
  end;

var
  frmFloatEdit1: TfrmFloatEdit1;

implementation

{$IFDEF CLR}
{$R *.NFM}
{$ELSE}
{$R *.DFM}
{$ENDIF}


procedure TfrmFloatEdit1.Button1Click(Sender: TObject);
begin
  Z1 := MtxFloatEdit1.CplxPosition;
  Z2 := MtxFloatEdit2.CplxPosition;
  if Sender = Button1 then Res := Math387.CMul(Z1,Z2)
  else if Sender = Button2 then Res := Math387.CDiv(Z1,Z2)
  else if Sender = Button3 then Res := Math387.CAdd(Z1,Z2)
  else if Sender = Button4 then Res := Math387.CSub(Z1,Z2);
  Edit1.Text := CplxToStr(Res,4);
end;

procedure TfrmFloatEdit1.Button5Click(Sender: TObject);
begin
  MtxFloatEdit1.Execute(true);
end;

procedure TfrmFloatEdit1.Button6Click(Sender: TObject);
begin
  MtxFloatEdit2.Execute(true);
end;

procedure TfrmFloatEdit1.FormCreate(Sender: TObject);
begin
  inherited;

  With RichEdit1.Lines, RichEdit1 do
  begin
    Add('The TMtxFloatEdit component can be used to edit/view/change '
    + 'real or complex number. The control also includes runtime complex '
    + 'number format editor. You can invoke it with :');

    Paragraph.FirstIndent := 10;
    Paragraph.Numbering := nsBullet;
    Add('holding the CTRL key and double clicking on TMtxFloatEdit control');
    Add('in code by calling TMtxFloatEdit Execute method'+#10+#13);
    Paragraph.FirstIndent := 0;
    Paragraph.Numbering := nsNone;
  end;

  MtxFloatEdit1.Value := '2+2i';
  MtxFloatEdit2.Value := '2+5i';
end;

initialization
  RegisterClass(TfrmFloatEdit1);

end.

File: MtxVec.CPP/FloatEdit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FloatEdit1.h"
#include "MtxExpr.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Basic2"
#pragma link "MtxComCtrls"
#pragma resource "*.dfm"
TfrmFloatEdit1 *frmFloatEdit1;
//---------------------------------------------------------------------------
__fastcall TfrmFloatEdit1::TfrmFloatEdit1(TComponent* Owner)
  : TBasicForm2(Owner)
{
  AnsiString str;

  str = "The TMtxFloatEdit component can be used to edit/view/change ";
  str+= "real or complex number. The control also includes runtime complex ";
  str+= "number format editor. You can invoke it with :";
  RichEdit1->Lines->Add(str);
  RichEdit1->Paragraph->FirstIndent = 10;
  RichEdit1->Paragraph->Numbering = nsBullet;
  RichEdit1->Lines->Add("holding the CTRL key and double clicking on TMtxFloatEdit control");
  RichEdit1->Lines->Add("in code by calling TMtxFloatEdit Execute method");
  RichEdit1->Paragraph->FirstIndent = 0;
  RichEdit1->Paragraph->Numbering = nsNone;

  MtxFloatEdit1->Value = "2+2i";
  MtxFloatEdit2->Value = "2+5i";
}
//---------------------------------------------------------------------------

void __fastcall TfrmFloatEdit1::Button1Click(TObject *Sender)
{
  Z1 = MtxFloatEdit1->CplxPosition;
  Z2 = MtxFloatEdit2->CplxPosition;
  if (Sender == Button1)
  {
	 Res = CMul(Z1, Z2);
  }
  else if (Sender == Button2) Res = CDiv(Z1,Z2);
  else if (Sender == Button3) Res = CAdd(Z1,Z2);
  else if (Sender == Button4) Res = CSub(Z1,Z2);
  Edit1->Text = CplxToStr(Res,2);
}
//---------------------------------------------------------------------------

void __fastcall TfrmFloatEdit1::Button5Click(TObject *Sender)
{
  MtxFloatEdit1->Execute(true);
}
//---------------------------------------------------------------------------

void __fastcall TfrmFloatEdit1::Button6Click(TObject *Sender)
{
  MtxFloatEdit2->Execute(true);
}
//---------------------------------------------------------------------------


